### PR TITLE
[MCHANGES-369] Add onlyCurrentVersion for announcement-generate

### PR DESCRIPTION
**Pull request by slawekjaranowski**
https://issues.apache.org/jira/browse/MCHANGES-369

I will not create a new test for it ...

Many classes need to refactor for easy (or possibility) testing ...

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/changes/announcement/AnnouncementMojo.java [hunk 1]
```diff
@@ -665,6 +676,12 @@ protected List<Release> getJiraReleases() throws MojoExecutionException {
                         + " that matched the versionPrefix '" + versionPrefix + "'.");
             }
 
+            if (onlyCurrentVersion) {
+                String version = (versionPrefix == null ? "" : versionPrefix) + project.getVersion();
+                issueList = IssueUtils.getIssuesForVersion(issueList, version);
+                getLog().info("The JIRA Report will contain issues only for the current version.");
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/changes/announcement/AnnouncementMojo.java:682]: probably should be debug level

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/plugins/changes/announcement/AnnouncementMojo.java:682]: It is the same in jira report, by the way this should be also refactored .... 
`jiraDownloader` should does such work

Comment (elharo) [reply src/main/java/org/apache/maven/plugins/changes/announcement/AnnouncementMojo.java:682]: I'm not sure I follow. Is this message going directly into the report or is it on the console when the report generation runs?

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/plugins/changes/announcement/AnnouncementMojo.java:682]: on console, changed to debug

**Review by elharo — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #240

**Pull request merged**